src/packets.rs

```diff
@@ -135,7 +135,7 @@ pub struct PingReq;
 pub struct PingResp;
 
 /// An MQTT PUBACK control packet
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Deserialize, Serialize)]
```

Comment:
Need to test serialization of PUBACKs

Reply:
Unit testing actually shows that this packet is being serialized improperly because of deviating definitions for serialized vs. deserialized properties. I think we need to reimagine what properties look like before landing this.